### ISSUE TITLE
Row length out of range

### ISSUE THREAD

**Issue by SaminSid** [openlink/virtuoso-opensource#493]
Hi,
I'm getting error GPF: page.c: 119 row length out of range whenever I try to start virtuoso. I'm using this db http://elba.informatik.uni-freiburg.de/freebase-qa/data/

Regards,
Samin

**Comment by HughWilliams** [comment]
@SaminSid: I assume you are referring to http://elba.informatik.uni-freiburg.de/freebase-qa/data/virtuoso.tar.gz  ?

What Operating system are you using and what is the version of Virtuoso being used ? Please provide the output of running  "virtuoso-t -?" which outputs version details ...

**Comment by SaminSid** [comment]
I'm using linux(32 bits ubuntu 14.04.3), virtuoso 7/develop version.

This is the last lines of output.
00   SUCCESS plugin 3: loaded from /home/samin/Downloads/aqqu-master/virtuoso-opensource/virtuoso/install/lib/virtuoso/hosting/creolewiki.so }
15:10:00 OpenLink Virtuoso Universal Server
15:10:00 Version 06.01.3127-pthreads for Linux as of Oct 31 2015
15:10:00 uses parts of OpenSSL, PCRE, Html Tidy
15:10:01 Database version 3126
15:10:01 SQL Optimizer enabled (max 1000 layouts)
15:10:02 Compiler unit is timed at 0.000786 msec
15:10:03 install/bin/virtuoso-t() [0x8467c99]
15:10:03 install/bin/virtuoso-t() [0x8467d0f]
15:10:03 install/bin/virtuoso-t() [0x80d3a43]
15:10:03 install/bin/virtuoso-t() [0x80aaf0e]
15:10:03 install/bin/virtuoso-t() [0x80704a0]
15:10:03 install/bin/virtuoso-t() [0x80c3290]
15:10:03 install/bin/virtuoso-t() [0x847018b]
15:10:03 /lib/i386-linux-gnu/libpthread.so.0(+0x6f70) [0xb752cf70]
15:10:03 /lib/i386-linux-gnu/libc.so.6(clone+0x5e) [0xb7418bee]
15:10:03 GPF: page.c:119 row length out of range
GPF: page.c:119 row length out of range

**Comment by HughWilliams** [comment]
@SaminSid: As the log clearly shows you are using Virtuoso 06.01.3127 and not a build from the virtuoso 7/develop git tree whose version would be 07.20.3214 ???

Looking at the build date of the binary in the log it would appear you compiled it yourself in which case you must have checkout a 6/develop or stable git branch . Run the "git status" command to see which tree is being used ...

**Comment by SaminSid** [comment]
Sorry yeah I'm using 6/develop version because I was getting following issue on stable/develop build/7 on ubuntu(14.04.3)
https://github.com/openlink/virtuoso-opensource/issues/160

**Comment by HughWilliams** [comment]
@SaminSid: I have a Ubuntu 14.04.3 VM and can build develop/7 and use if fine, the git#160 issue was also closed back in 2014 when it was resolved .

You need to use a Virtuoso 7.2 build with that Freebase preloaded database as that is what is was created with looking at the engine number on startup.

Otherwise you have to load the raw RDF datasets into a version of your choice ...

**Comment by SaminSid** [comment]
I'm getting this error while trying to configure on linux(32 bits). Now I'm using develop/7.
CFLAGS="-O2"
export CFLAGS
./configure 
checking for long double with more range or precision than double... yes
checking size of char... 1
checking size of short... 2
checking size of unsigned... 4
checking size of int... 4
checking size of long... 4
checking size of long long... 8
checking size of float... 4
checking size of double... 8
checking size of long double... 12
checking size of char *... 4
checking size of void *... 4
<b>configure: error: The current version of Virtuoso Open Source Edition (Column Store) can only be build on 64bit platforms</b>

**Comment by HughWilliams** [comment]
@SaminSid: That is correct Virtuoso 7 is only supported on 64bit platforms, can you not use a 64bit Linux OS i.e. is your host machine really 32bit still ? If so then you would have to use Virtuoso 6 and load those freebase RDF datasets yourself into an empty Virtuoso 6 database instance ...

**Comment by SaminSid** [comment]
I was finally able to load virtuoso server(develop/7 on linux-64) with empty db. 
But when I try to load http://elba.informatik.uni-freiburg.de/freebase-qa/data/ I'm getting lck file cannot be created error. Any ideas??

**Comment by HughWilliams** [comment]
@SaminSid: Please provide detailed on the steps being performed and any errors occurring as I see no relation between and lck which gets created when the database is started and the load for RDF datasets into the database presumably with the Virtuoso RDF Bulk Loader ?

**Comment by SaminSid** [comment]
This is where virtuoso works..
db$ sudo virtuoso-t -f

Sat Nov 07 2015
07:52:21 { Loading plugin 1: Type `plain', file`wikiv' in `/home/samin/Downloads/aqqu-master/virtuoso-opensource/virtuso/install/lib/virtuoso/hosting'
07:52:21   FAILED  plugin 1: Unable to locate file }
07:52:21 { Loading plugin 2: Type`plain', file `mediawiki' in`/home/samin/Downloads/aqqu-master/virtuoso-opensource/virtuso/install/lib/virtuoso/hosting'
07:52:21   FAILED  plugin 2: Unable to locate file }
07:52:21 { Loading plugin 3: Type `plain', file`creolewiki' in `/home/samin/Downloads/aqqu-master/virtuoso-opensource/virtuso/install/lib/virtuoso/hosting'
07:52:21   FAILED  plugin 3: Unable to locate file }
07:52:21 OpenLink Virtuoso Universal Server
07:52:21 Version 07.20.3214-pthreads for Linux as of Nov  4 2015
07:52:21 uses parts of OpenSSL, PCRE, Html Tidy
07:52:21 Database version 3126
07:52:21 SQL Optimizer enabled (max 1000 layouts)
07:52:24 Compiler unit is timed at 0.000154 msec
07:52:41 Roll forward started
07:52:41 Roll forward complete
07:52:41 Checkpoint started
07:52:41 Checkpoint finished, log reused
07:52:42 HTTP/WebDAV server online at 8890
07:52:42 Server online at 1111 (pid 2355)

I'm not using RDF Bulk loader I'm trying to use virtuoso.db directly from this link http://elba.informatik.uni-freiburg.de/freebase-qa/data/

**Comment by HughWilliams** [comment]
@SaminSid: So where is the problem with the lck file as the log you provide shows the Virtuoso server has successfully started and is running on HTTP port 8890 and SQL port 1111 ???

**Comment by SaminSid** [comment]
I updated var/db/virtuoso.ini to point to <b>freebase-db</b> and it seems to be working now.
But when I tried to start virtuoso with  http://elba.informatik.uni-freiburg.de/freebase-qa/data/ virtuoso.ini but it was failing with following error.
Sat Nov 07 2015
07:54:31 Unable to create file virtuoso-db/virtuoso.lck (No such file or directory).
07:54:31 This probably means you either do not have permission to start
07:54:31 this server, or that virtuoso-t is already running.
07:54:31 If you are absolutely sure that this is not the case, please try
07:54:31 to remove the file virtuoso-db/virtuoso.lck and start again.

**Comment by HughWilliams** [comment]
Well the problem is self evident, so you have a "virtuoso-db" sub directory in directory when Virtuoso is being run from ? 

Clearly not hence the error and if you look at the INI file provided by whoever created the tar ball it contains:

```
DatabaseFile                    = virtuoso-db/virtuoso.db
ErrorLogFile                    = virtuoso-db/virtuoso.log
LockFile                        = virtuoso-db/virtuoso.lck
TransactionFile                 = virtuoso-db/virtuoso.trx
xa_persistent_file              = virtuoso-db/virtuoso.pxa
```

Which is why when you use it the server tries to create the lck file in that location ...

So u either use your existing INI file (suitably tuned) as you have done or edit the INI file provided and make sure any locations (relative or absolute) exist on your machine ...
